### PR TITLE
NO-ISSUE: Make setup_env user friendly

### PR DESCRIPTION
**Pull request by celebdor**
# Assisted Pull Request

## Description

Up until now, setup_env has always tried to install things at a system
level. With this, it will install things in ~/.local/bin if:
- Running in fedora toolbox
- Running on a systemd distribution as a non-root user

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

Tested:
Root outside of container (Fedora silverblue) -> Installs executables to /usr/local/bin
Root inside Fedora toolbox container (Fedora silverblue) -> Installs executables to /usr/local/bin
Regular user outside of container (Fedora silverblue) -> Installs executables to ${HOME}/.local/bin
Regular inside Fedora toolbox container (Fedora silverblue) -> Installs executables to ${HOME}/.local/bin

## Assignees

/cc @lranjbar 
/cc @osherdp 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

### CODE REVIEW

File: hack/setup_env.sh [hunk 1]
```diff
@@ -4,45 +4,62 @@ set -o nounset
...
+  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o "${tmpdir}/golang_installer"
+  chmod u+x ${tmpdir}/golang_installer
+  "${tmpdir}/golang_installer"
   rm /tmp/golang_installer
+  rmdir "$tmpdir"
 
   echo "Activating go command on current shell..."
   set +u
-  source /root/.bash_profile
+  source "${HOME}/.bash_profile"
```
Comment (flaper87) [inline hack/setup_env.sh:47]: This reminds me that I have a pending PR for the `deploy/operator` scripts to do the same.

File: hack/setup_env.sh [hunk 2]
```diff
@@ -4,45 +4,62 @@ set -o nounset
...
-  curl -L --retry 5 "https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv4.3.0/kustomize_v4.3.0_linux_amd64.tar.gz" | \
-    tar -zx -C /usr/bin/
 }
 
 function golang() {
   echo "Installing golang..."
-  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o /tmp/golang_installer
-  chmod u+x /tmp/golang_installer
-  /tmp/golang_installer
+  local tmpdir="$(mktemp -d)"
```
Comment (flaper87) [inline hack/setup_env.sh:38]:
```suggestion
  local tmpdir="$(mktemp -d 2>/dev/null || mktemp -d -t 'mytmpdir')"
```

If we want this to be OSx compliant

File: hack/setup_env.sh [hunk 3]
```diff
@@ -4,45 +4,62 @@ set -o nounset
...
+    if systemd-detect-virt -cq; then
+        if [ -n "${TOOLBOX_PATH+x}" ]; then
+            BINDIR="$(systemd-path user-binaries)"
+        fi
+    else
+        BINDIR="$(systemd-path user-binaries)"
+    fi
+else
+    BINDIR="/usr/local/bin"
+fi
```
Comment (flaper87) [inline hack/setup_env.sh:21]:
```suggestion
if command -v systemd-detect-virt; then
    if systemd-detect-virt -cq; then
        if [ -n "${TOOLBOX_PATH+x}" ]; then
            BINDIR="$(systemd-path user-binaries)"
        fi
    else
        BINDIR="$(systemd-path user-binaries)"
    fi
else
    BINDIR="/usr/local/bin"
fi
```

**Review by flaper87 — CHANGES_REQUESTED** [review]
small nits here and there.

File: hack/setup_env.sh (hunk 1 above)
Comment (celebdor) [reply hack/setup_env.sh:47]: Oops! I would have sworn I did this part. Or maybe it was the one that made me decide to finish it in the morning

File: hack/setup_env.sh (hunk 2 above)
Comment (YuviGold) [reply hack/setup_env.sh:38]: @flaper87  Out of curiosity, why would the original code fail?

File: hack/setup_env.sh [hunk 4]
```diff
@@ -4,45 +4,62 @@ set -o nounset
...
   echo "Installing golang..."
-  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o /tmp/golang_installer
-  chmod u+x /tmp/golang_installer
-  /tmp/golang_installer
+  local tmpdir="$(mktemp -d)"
+  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o "${tmpdir}/golang_installer"
+  chmod u+x ${tmpdir}/golang_installer
+  "${tmpdir}/golang_installer"
   rm /tmp/golang_installer
+  rmdir "$tmpdir"
```
Comment (YuviGold) [inline hack/setup_env.sh:43]:
```suggestion
  rmdir "${tmpdir}"
```

**Comment by flaper87** [comment]
@celebdor will you have time to address the comments here? Or should we close it for now?

**Comment by flaper87** [comment]
Closing it for now as PR has been in a stale state for a while
/close

**Pull request closed without merging**